common/kokkos-sampler/kp_sampler_skip.cpp

```diff
@@ -0,0 +1,189 @@
+#include <cstdio>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <dlfcn.h>
+
+static uint64_t uniqID = 0;
+static uint64_t kernelCounter = 0;
+static uint64_t kernelSampleSkip = 101;
+static int tool_verbosity;
+
+typedef void (*initFunction)(const int, const uint64_t, const uint32_t, void*);
+typedef void (*finalizeFunction)();
+typedef void (*beginFunction)(const char*, const uint32_t, uint64_t*);
+typedef void (*endFunction)(uint64_t);
+
+static initFunction initProfileLibrary = NULL;
+static finalizeFunction finalizeProfileLibrary = NULL;
+static beginFunction beginForCallee = NULL;
+static beginFunction beginScanCallee = NULL;
+static beginFunction beginReduceCallee = NULL;
+static endFunction endForCallee = NULL;
+static endFunction endScanCallee = NULL;
+static endFunction endReduceCallee = NULL;
+
+extern "C" void kokkosp_init_library(const int loadSeq,
+	const uint64_t interfaceVer,
+	const uint32_t devInfoCount,
+	void* deviceInfo) {
+
+	const char* tool_verbose_str = getenv("KOKKOS_SAMPLER_VERBOSE");
+	if( NULL != tool_verbose_str ) {
+		tool_verbosity = atoi( tool_verbose_str );
+	} else {
+		tool_verbosity = 0;
+	}
+
+	char* profileLibrary = getenv("KOKKOS_PROFILE_LIBRARY");
```

Comment:
This needs to check for KOKKOS_TOOLS_LIBS or both. And I would suggest error checks on getting nullptr back in case this wasn't set. For example someone could in theory compile the tool into their executable and use the callback setter interface to leverage this. In which case no environment variable is set or needs to be set.

Reply:
Thanks. Your revised code - with The warning of Kokkos_PROFILE_Library being deprecated - seems to addresses the comment.

I think you also may want to emit a warning - or note to the user - for the case of a null pointer where the tool is built with the executable. You would need to do this done only if the behavior of the tool when it is directly linked with the application program's (which includes performance and output of the logging) is significantly different than the case when the user specifies Kokkos_TOOLS_LIB environment variable. This may be nitpicking, but I wanted to make sure to point this out if helpful.

This looks good to me as it is.